File: Appium_C#_Test/KotlinTrainingTestProject/Capabilities.cs
using Microsoft.VisualStudio.TestTools.UnitTesting;
using OpenQA.Selenium;
using OpenQA.Selenium.Appium;
using OpenQA.Selenium.Appium.Android;
using OpenQA.Selenium.Remote;
using System;
using TechTalk.SpecFlow;

namespace KotlinTrainingTestProject
{
    public abstract class Capabilities
    {
        protected static AppiumDriver<IWebElement> driver;

        [BeforeScenario]
        [TestInitialize]
        public void Initialyze()
        {
            GetCapabilities();
        }

        [TestCleanup]
        [AfterScenario]
        public void AfterTestRun()
        {
            driver.Quit();
        }

        protected static void GetCapabilities()
        {
            var cap = new DesiredCapabilities();
            cap.SetCapability("deviceName", "pixel_api_25");
            cap.SetCapability("appActivity", "activities.MainActivity");
            cap.SetCapability("appPackage", "com.example.jpedretti.kotlintraining");
            cap.SetCapability("clearSystemFiles", true);

            driver = new AndroidDriver<IWebElement>(new Uri("http://127.0.0.1:4723/wd/hub"), cap);
        }

        protected static By ByText(string text) => new ByAndroidUIAutomator($"new UiSelector().textContains(\"{text}\")");
    }
}

File: Appium_Java_Test/src/test/java/com/example/jpedretti/kotlintraining/java/test/Capabilities.kt
package com.example.jpedretti.kotlintraining.java.test

import io.appium.java_client.AppiumDriver
import io.appium.java_client.android.AndroidDriver
import org.openqa.selenium.WebElement
import org.openqa.selenium.remote.DesiredCapabilities
import java.net.URL

open class Capabilities {
    lateinit var driver: AppiumDriver<WebElement>

    fun setDeviceCapabilities() {
        // File classpathRoot = new File(System.getProperty("user.dir"));
        // File appDir = new File(classpathRoot, "../../../apps/ApiDemos/bin");
        // File app = new File(appDir.getCanonicalPath(), "ApiDemos-debug.apk");
        val capabilities = DesiredCapabilities()
        capabilities.setCapability("deviceName", "pixel_api_25")
        // capabilities.setCapability("app", app.getAbsolutePath());
        capabilities.setCapability("appPackage", "com.example.jpedretti.kotlintraining")
        capabilities.setCapability("appActivity", ".activities.MainActivity")
        capabilities.setCapability("clearSystemFiles", true)
        driver = AndroidDriver(URL("http://localhost:4723/wd/hub"), capabilities)
    }
}